sysdump/sysdump.go

```diff
@@ -180,6 +181,43 @@ func Collect(c *k8s.Client, o Options) error {
 		}
 		return nil
 	})
+	// Saves the probe data in the zip file
+	errs.Go(func() error {
+		oldStdOut := os.Stdout
+		reader, writer, err := os.Pipe()
+		if err != nil {
+			return err
+		}
+		os.Stdout = writer
+		err = probe.PrintProbeResult(c, probe.Options{
+			Namespace: "",
+			Full:      false,
+			Output:    "no-color",
+			GRPC:      "",
+		})
+		if err != nil {
+			return err
+		}
+		err = writer.Close()
+		if err != nil {
+			return err
+		}
+		os.Stdout = oldStdOut
+		out, _ := io.ReadAll(reader)
+		outStr := string(out)
+		outArr := strings.Split(outStr, "\nFound KubeArmor running in Kubernetes\n\n")
```

Comment:
why are we doing this?

Reply:
There were some absurds strings getting printed like `getting logs from pod name` and after your suggestion I thought of removing it.

Reply:
imo we shouldn't handle it like this, it might require flushing the os.stdout before writing the probe output or may be locks so all the go routines don't write in stdout at the same time.  @Prateeknandle @DelusionalOptimist  wdyt ?

Reply:
I also had some doubt in handling concurrency. I think using locks would be helpful but then we have to add them everywhere else in the codebase. But we can flush OS.Stdout for sure!

Reply:
So after the meet we got to the conclusion that we should not use locks as it might block other go routines! What I have thought as a solution is using a different writer by creating and passing it in `print` function. It will fix that as no longer we will have to depend on `os.Stdout`